event_model/__init__.py

```diff
@@ -80,47 +80,79 @@ def _dispatch(self, name, doc, validate):
         Optionally validate that the result is still a valid document.
         """
         output_doc = getattr(self, name)(doc)
+
+        # If 'event' is not defined by the subclass but 'event_page' is, or
+        # vice versa, use that. And the same for 'datum_page' / 'datum.
+        if output_doc is NotImplemented:
+            if name == 'event':
+                event_page = pack_event_page(doc)
+                # Subclass' implementation of event_page may return a valid
+                # EventPage or None or NotImplemented.
+                output_event_page = self.event_page(event_page) or event_page
+                if output_event_page is not NotImplemented:
+                    output_doc, = unpack_event_page(output_event_page)
+            elif name == 'datum':
+                datum_page = pack_datum_page(doc)
+                # Subclass' implementation of datum_page may return a valid
+                # DatumPage or None or NotImplemented.
+                output_datum_page = self.datum_page(datum_page) or datum_page
+                if output_datum_page is not NotImplemented:
+                    if output_datum_page is None:
```

Comment:
This line doesn't appear in the if name == 'event' case

Reply:
Thanks. Note to self: the fix is to remove it here. Using `or datum_page` above handles it and guarantees that `output_datum_page` will not be `None`.